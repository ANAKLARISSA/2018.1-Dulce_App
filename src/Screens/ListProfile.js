import React, {Component} from 'react';
import {View} from 'react-native';
import {Container, Content, Card, CardItem, Text, Body, List, Fab, Icon} from 'native-base';
import ScreenHeader from '../Components/ScreenHeader';

const styles = {
  center: {
    flex: 1,
    flexDirection: 'column',
    alignItems: 'center',
    marginLeft: 68,
    justifyContent: 'center'
  },

  colorPurple: {
    color: '#5f4b8b',
    fontSize: 17,
    marginLeft: 5,
    fontFamily: 'Raleway'
  }
};
export default class ListProfile extends Component {

  constructor(props) {
    super(props);
    this.state = {
      notifications: [
        {
          name: 'Hospital Regional Do Gama',
          matricula: '123456',
          setor: 'Jimmy'
        },{
          name: 'Hospital Regional da Asa Norte',
          matricula: '160119553',
          setor: 'Fepas Beauty'
        }
      ]
    };
  }

  fab() {
    return (
    <Fab
       active={true}
      direction='up'
      containerStyle={{}}
      style={{backgroundColor: '#5f4b8b'}}
      position='bottomRight'
        onPress={() => {}}>
        <Icon type='MaterialIcons' name='add' />
    </Fab>);
  }

  renderCard(item) {
    return (
      <Card>
        <CardItem header><Text style={styles.colorPurple}>{item.name}</Text></CardItem>
        <CardItem style={styles.center} button onPress={() => {}}>
          <Body>
            <View >
              <Text> Setor: <Text style={styles.colorPurple}>{item.setor}</Text></Text>
              <Text> Matricula: <Text style={styles.colorPurple}>{item.matricula}</Text></Text>
            </View>
          </Body>
        </CardItem>
      </Card>
    );
  }

  render() {
    return (
      <View style={{flex: 1}}>
        <Container>
          <ScreenHeader
            title = 'Lista de Perfis'/>
          <Content>
            <List dataArray={this.state.notifications}
              renderRow={(item) => (
                this.renderCard(item))} />
          </Content>
        </Container>
        {this.fab()}
      </View>
    );
  }
}
