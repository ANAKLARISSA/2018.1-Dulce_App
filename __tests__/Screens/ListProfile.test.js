import React from 'react';
import renderer from 'react-test-renderer';
import ListProfile from '../../src/Screens/ListProfile';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
    .create(<ListProfile />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('testing Fab\'s onPress', () => {
  const navigation = {navigate: jest.fn()};
  const wrapper = shallow(<ListProfile navigation={navigation} />);
  console.log(wrapper.debug());
  const fab = wrapper.find('Styled(Fab)');
  fab.simulate('press');
  expect(navigation.navigate).toHaveBeenCalled();
});
