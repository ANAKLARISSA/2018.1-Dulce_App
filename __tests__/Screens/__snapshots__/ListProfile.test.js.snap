// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fd7d01",
        "flex": 0.25,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f1f1f1",
          "borderRadius": 5,
          "height": 40,
          "margin": 1,
          "width": 7.5,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 8,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "height": 1334,
        }
      }
      virtual={undefined}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#F8F8F8",
                "borderBottomColor": "#a7a6ab",
                "borderBottomWidth": 0.5,
                "elevation": 3,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 15,
                "right": 0,
                "shadowColor": undefined,
                "shadowOffset": undefined,
                "shadowOpacity": undefined,
                "shadowRadius": undefined,
                "top": 0,
              },
              Object {
                "backgroundColor": "#5f4b8b",
              },
            ]
          }
          virtual={undefined}
        >
          <View />
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 17,
                    "fontWeight": "600",
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "Raleway",
                  },
                ]
              }
              virtual={undefined}
            >
               
              Perfis
               
            </Text>
          </View>
        </View>
      </View>
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
          }
        }
        viewIsInsideTabBar={false}
        virtual={undefined}
      >
        <View>
          <RCTScrollView
            closeOnRowBeginSwipe={false}
            closeOnRowPress={true}
            closeOnScroll={true}
            dataArray={
              Array [
                Object {
                  "matricula": "123456",
                  "name": "Hospital Regional Do Gama",
                  "setor": "Jimmy",
                },
                Object {
                  "matricula": "160119553",
                  "name": "Hospital Regional da Asa Norte",
                  "setor": "Fepas Beauty",
                },
              ]
            }
            dataSource={
              ListViewDataSource {
                "items": 2,
              }
            }
            directionalDistanceChangeThreshold={2}
            disableLeftSwipe={false}
            disableRightSwipe={false}
            enableEmptySections={true}
            leftOpenValue={0}
            previewFirstRow={false}
            recalculateHiddenLayout={false}
            renderRow={[Function]}
            renderScrollComponent={[Function]}
            rightOpenValue={0}
            style={Object {}}
            swipeToOpenPercent={50}
            virtual={undefined}
          >
            <View>
              <View
                iconLeft={true}
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ccc",
                    "borderLeftWidth": 0.5,
                    "borderRadius": 2,
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                    "elevation": 3,
                    "flex": 1,
                    "flexWrap": "nowrap",
                    "marginBottom": 5,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 5,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 1.5,
                  }
                }
                virtual={undefined}
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 2,
                      "flex": 1,
                      "flexDirection": "column",
                      "marginLeft": 2,
                      "opacity": 1,
                      "paddingBottom": 10,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 10,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "alignSelf": "stretch",
                        "flex": 1,
                      }
                    }
                    virtual={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#5f4b8b",
                            "fontFamily": "Raleway",
                            "fontSize": 17,
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      Hospital Regional Do Gama
                    </Text>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 12,
                            },
                            Array [
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "#5f4b8b",
                                "fontSize": 40,
                                "padding": 10,
                              },
                            ],
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                        type="MaterialCommunityIcons"
                        virtual={undefined}
                      >
                        
                      </Text>
                      <View
                        style={
                          Object {
                            "marginTop": 10,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            }
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                           Setor: 
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#5f4b8b",
                                  "fontFamily": "Raleway",
                                  "fontSize": 17,
                                },
                              ]
                            }
                            uppercase={false}
                            virtual={undefined}
                          >
                            Jimmy
                          </Text>
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            }
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                           Matricula: 
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#5f4b8b",
                                  "fontFamily": "Raleway",
                                  "fontSize": 17,
                                },
                              ]
                            }
                            uppercase={false}
                            virtual={undefined}
                          >
                            123456
                          </Text>
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
              <View
                iconLeft={true}
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#ccc",
                    "borderLeftWidth": 0.5,
                    "borderRadius": 2,
                    "borderRightWidth": 0.5,
                    "borderTopWidth": 0.5,
                    "elevation": 3,
                    "flex": 1,
                    "flexWrap": "nowrap",
                    "marginBottom": 5,
                    "marginLeft": 2,
                    "marginRight": 2,
                    "marginTop": 5,
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                    "shadowOpacity": 0.1,
                    "shadowRadius": 1.5,
                  }
                }
                virtual={undefined}
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#fff",
                      "borderRadius": 2,
                      "flex": 1,
                      "flexDirection": "column",
                      "marginLeft": 2,
                      "opacity": 1,
                      "paddingBottom": 10,
                      "paddingLeft": 15,
                      "paddingRight": 15,
                      "paddingTop": 10,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "flex-start",
                        "alignSelf": "stretch",
                        "flex": 1,
                      }
                    }
                    virtual={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#000",
                            "fontFamily": "System",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#5f4b8b",
                            "fontFamily": "Raleway",
                            "fontSize": 17,
                          },
                        ]
                      }
                      uppercase={false}
                      virtual={undefined}
                    >
                      Hospital Regional da Asa Norte
                    </Text>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": undefined,
                              "fontSize": 12,
                            },
                            Array [
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "#000",
                                "fontSize": 30,
                              },
                              Object {
                                "color": "#5f4b8b",
                                "fontSize": 40,
                                "padding": 10,
                              },
                            ],
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                        type="MaterialCommunityIcons"
                        virtual={undefined}
                      >
                        
                      </Text>
                      <View
                        style={
                          Object {
                            "marginTop": 10,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            }
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                           Setor: 
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#5f4b8b",
                                  "fontFamily": "Raleway",
                                  "fontSize": 17,
                                },
                              ]
                            }
                            uppercase={false}
                            virtual={undefined}
                          >
                            Fepas Beauty
                          </Text>
                        </Text>
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          ellipsizeMode="tail"
                          style={
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            }
                          }
                          uppercase={false}
                          virtual={undefined}
                        >
                           Matricula: 
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "#000",
                                  "fontFamily": "System",
                                  "fontSize": 16,
                                },
                                Object {
                                  "color": "#5f4b8b",
                                  "fontFamily": "Raleway",
                                  "fontSize": 17,
                                },
                              ]
                            }
                            uppercase={false}
                            virtual={undefined}
                          >
                            160119553
                          </Text>
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
      </RCTScrollView>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "alignItems": "center",
          "bottom": 20,
          "flexDirection": "column",
          "height": 56,
          "left": undefined,
          "position": "absolute",
          "right": 20,
          "top": undefined,
          "width": 56,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5f4b8b",
            "borderRadius": 28,
            "bottom": 0,
            "elevation": 4,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "position": "absolute",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 2,
            "width": 56,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#fff",
                  "fontSize": 24,
                  "lineHeight": 27,
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
          type="MaterialIcons"
          virtual={undefined}
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
